docs/DeveloperGuide.adoc

```diff
@@ -278,7 +278,88 @@ are applied when restoring `**THRIFT**` data from the data file.
+
+Step 1. The user launches the application with data from previous sessions. THRIFT does contains only one `**Transaction**` containing no tags and the user is going to perform tagging and un-tagging operations on it.
+
+image::tag-untag/TagUntagTransactionWoTag.png[]
+
+Step 2. The user executes the command **tag i/1 t/Delicious** to tag the `**Transaction**` at `**Index**` 1 with the `**Tag**` "Delicious". The input is checked with a `**TagCommandParser**` parser and an attempt to parse each parameter occurs:
+
+* `**Index**` is parsed by `ParserUtil#parseIndex(String)`
+* `**Tag**` strings are parsed by `ParserUtil#parseTags(Collection<String>)` which calls `ParserUtil#parseTag(String)` iteratively for every string in the collection
+
+TIP: `**ParserUtil**` is a class that contains useful functions for parsing the inputs from the user
+
+[NOTE]
+Tags that already exist inside the specified `**Transaction**` will be ignored, and if that results in no tags being updated, an error will be shown to the user.
+
+Since the user input is valid, a `**TagCommand**` command object is created and executed. As a result, a copy of the `**Transaction**` object with the new set of `**Tag**` objects with the specified `**Tag**` inside replaces the original in the `**TransactionList**` at the same position.
+
+image::tag-untag/TagUntagTransactionWTag.png[]
+
+Step 3. The user realises that the Tag added was not appropriate. The user then executes the command **untag i/1 t/Delicious** to untag the `**Tag**` "Delicious" from the `**Transaction**` at `**Index**` 1. The input is now checked with a `**UntagCommandParser**` parser and similarly, an attempt to parse each parameter occurs.
+
+* `**Index**` is parsed by `ParserUtil#parseIndex(String)`
+* `**Tag**` strings are parsed by `ParserUtil#parseTags(Collection<String>)` which calls `ParserUtil#parseTag(String)` iteratively for every string in the collection
+
+[NOTE]
+Tags that do not exist inside the specified `**Transaction**` will be ignored, and if that results in no tags being updated, an error will be shown to the user.
+
+Again, since the user input is valid, a `**UntagCommand**` command object is created and executed. As a result, a copy of the `**Transaction**` object with the new set of `**Tag**` objects with the specified `**Tag**` removed replaces the original in the `**TransactionList**` at the same position.
+
+image::tag-untag/TagUntagTransactionWoTag.png[]
+
+//The following activity diagram summarises what happens the user executes a command to **tag** a `**Transaction**`. The **untag** command follows the same sequence.
+
+
+==== Design Considerations
+
+===== Aspect: Mutability of `**Transaction**` objects
+
+* **Alternative 1: (current choice)**: Create a copy of the `**Transaction**` object with modified tags, and replace the original in the `**TransactionList**`
```

Comment:
... object with modified tags and replace ...

Reply:
Removed the comma.